.github/workflows/code-checks.yml

```diff
@@ -34,36 +34,17 @@ jobs:
       with:
         python-version: '3.10.6'
 
-    # Note: This uses an internal pip API and may not always work
-    # https://github.com/actions/cache/blob/master/examples.md#multiple-oss-in-a-workflow
-    - name: Cache pip
+    - name: Mypy cache
       uses: actions/cache@v3
       with:
-        path: ~/.cache/pip
-        key: pip-${{ hashFiles('requirements/pytorch/base.txt') }}
-        restore-keys: pip-
-        enableCrossOsArchive: true
+        path: .mypy_cache
+        key: mypy-${{ hashFiles('requirements/typing.txt') }}
 
     - name: Install dependencies
       run: |
-        pip install mypy==0.982  # TODO: pull this verion deom requirements/test.txt
-        pip install torch==1.12 --find-links https://download.pytorch.org/whl/cpu/torch_stable.html
-        python ./requirements/pytorch/adjust-versions.py requirements/pytorch/extra.txt
         # todo: adjust requirements for both code-bases
-        pip install \
-          -r requirements/fabric/base.txt \
-          -r requirements/pytorch/base.txt \
-          -r requirements/pytorch/extra.txt \
-          -r requirements/app/base.txt \
-          -r requirements/app/cloud.txt \
-          -r requirements/app/ui.txt \
```

Comment:
We need to install all these requirements still. Each of our dependencies might have their own typing definitions which we want our mypy workflow to use when checking

Reply:
So far I understood, instaling `types-*` is full replacement
but regarding these all `*/base.txt` are installed with this package so the remaining are 
- pytorch/extra.txt
- app/cloud.txt
- app/ui.txt

Reply:
The types change looks good.

Then those 3 should be kept

Reply:
addressed in [f4d1167](https://github.com/Lightning-AI/lightning/pull/16860/commits/f4d11675d1047e8f02931694d23298cee35f6fa8)